### PR TITLE
Add tests for the _jit_wrapper entry point

### PR DESCRIPTION
**Pull request by steven-johnson**
Add to example_aottest and user_context_aottest. (Issue #667)

### CODE REVIEW

**Comment by abadams** [comment]
I'm a little wary about changing the example aot test, because it's the minimal example we can point to for how to call into Halide as produced by a generator. Do you think it's worth moving this check to its own test?

**Comment by steven-johnson** [comment]
will do

**Comment by steven-johnson** [comment]
BTW: while I'm here, do we want to entertain giving it a less-weirdly-confusing name?

**Comment by steven-johnson** [comment]
renamed _jit_wrapper -> _argv (per Dillon's suggestion); added separate testcase and reverted the change to "example"; added prototype to generated header files. PTAL

File: test/generator/argvcall_aottest.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+int main(int argc, char **argv) {
+
+  int result;
+  Image<int32_t> output(kSize, kSize, 3);
+
+  // We can, of course, pass whatever values for Param/ImageParam that we like.
+  result = argvcall(1.2f, 3.4f, output);
+  if (result != 0) {
+    fprintf(stderr, "Result: %d\n", result);
```
Comment (abadams) [inline test/generator/argvcall_aottest.cpp]: Wrong indentation in this function

**Comment by abadams** [comment]
lgtm apart from the indentation nit

**Pull request merged**